pycqed/analysis_v2/base_analysis.py

```diff
@@ -470,6 +477,7 @@ def prepare_fitting(self):
         # initialize everything to an empty dict if not overwritten
         self.fit_dicts = OrderedDict()
 
+
     def run_fitting(self):
```

Comment:
Danger! I think this piece of code is getting awfully complicated to understand (as you have probably experienced when debugging the tests) I am sure the tests are working now but I'm a bit hesitant to approve this change... Let's discuss a bit, I'm not sure what the best solution is.